web-portal/pages/calendar.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="container info-page calendar-page">
+    <v-app>
+      <client-only>
+        <FullCalendar class="calendar" ref="fullCalendar" :options="calendarOptions" />
+      </client-only>
+    </v-app>
+  </div>
+</template>
+
+<script>
+  import { mapGetters } from 'vuex'
+  import FullCalendar from '@fullcalendar/vue'
+  import momentPlugin from '@fullcalendar/moment'
+  import dayGridPlugin from '@fullcalendar/daygrid'
+  import interactionPlugin from '@fullcalendar/interaction'
+  import timeGridPlugin from '@fullcalendar/timegrid'
+
+  export default {
+    components: {
+      FullCalendar
+    },
+    data: function () {
+      return {
+        timeInterval: null,
+        calendarOptions: {
+          plugins: [momentPlugin, dayGridPlugin, timeGridPlugin, interactionPlugin],
+          initialView: 'dayGridMonth',
+          nowIndicator: true,
+          editable: false,
+          headerToolbar: {
+            left: 'dayGridMonth,dayGridWeek,timeGridDay',
+            center: 'title',
+            right: 'prev,next'
+          },
+          titleFormat: { year: 'numeric', month: 'short', day: 'numeric' },
+          dateClick: this.handleDateClick,
+          eventClick: this.handleEventClick,
+          eventBackgroundColor: '#b7b09c',
+          eventBorderColor: '#fff',
+          eventTextColor: '#fff'
+        }
+      }
+    },
+    async fetch({ store }) {
+      await store.dispatch('LoadAllCalendarEventData')
+    },
+    computed: {
+      ...mapGetters({ getAllCalendarEvents: 'GetAllCalendarEvents' })
+    },
+    mounted() {
+      this.setUpdateCalendarEvents()
+
+      this.timeInterval = setInterval(() => {
+        this.$store.dispatch('LoadAllCalendarEventData').then(() => {
+          this.setUpdateCalendarEvents()
+        })
+      }, 5 * 60 * 1000)
+    },
+    methods: {
+      setUpdateCalendarEvents() {
+        const parsedEvents = []
+
+        this.getAllCalendarEvents.forEach((event) => {
+          event.date_times.forEach((eventDate) => {
+            parsedEvents.push({
+              title: event.title,
+              start: eventDate.start_time,
+              end: eventDate.end_time,
+              groupId: event.id,
+              editable: false,
+              originId: event.id
+            })
+          })
+        })
+
+        this.calendarOptions.events = parsedEvents
+      },
```

Comment:
Thoughts on refactoring this as a computed property? It would only be re-evaluated when the `getAllCalendarEvents` getter updated.
Would simplify the interval, too -- just need to fire off the store action and let Vue do the rest.

Reply:
@MatthewGidcomb good point! a3c6f98055a60de0dfb28d94cad1864740a024ed